### PR TITLE
plugin update fix

### PR DESCRIPTION
**Pull request by godrei**

### CODE REVIEW

File: configs/configs.go [hunk 1]
```diff
@@ -2,11 +2,20 @@ package configs
 
 import (
 	"path"
+	"time"
+
+	"gopkg.in/yaml.v2"
```
Comment (godrei) [inline configs/configs.go]: use json

File: configs/configs.go [hunk 2]
```diff
@@ -36,16 +45,55 @@ const (
 )
 
 const (
-	bitriseVersionSetupStateFileName = "setup.version"
+	bitriseConfigFileName = "config.yml"
```
Comment (godrei) [inline configs/configs.go]: config.json

File: configs/configs.go [hunk 3]
```diff
@@ -36,16 +45,55 @@ const (
...
 
-func getBitriseConfigVersionSetupFilePath() string {
-	return path.Join(GetBitriseConfigsDirPath(), bitriseVersionSetupStateFileName)
+func getBitriseConfigFilePath() string {
+	return path.Join(GetBitriseConfigsDirPath(), bitriseConfigFileName)
+}
+
+func loadBitriseConfig() (ConfigModel, error) {
+	if err := EnsureBitriseConfigDirExists(); err != nil {
+		return ConfigModel{}, nil
```
Comment (godrei) [inline configs/configs.go]: missing error handling

File: configs/configs.go [hunk 4]
```diff
@@ -36,16 +45,55 @@ const (
...
+	} else if !exist {
+		return ConfigModel{}, nil
+	}
+
+	bytes, err := fileutil.ReadBytesFromFile(configPth)
+	if err != nil {
+		return ConfigModel{}, err
+	}
+
+	if len(bytes) == 0 {
```
Comment (godrei) [inline configs/configs.go]: if file exists, it should not to be empty - remove this

File: configs/configs.go [hunk 5]
```diff
@@ -2,11 +2,20 @@ package configs
...
+	"gopkg.in/yaml.v2"
 
 	"github.com/bitrise-io/go-utils/fileutil"
 	"github.com/bitrise-io/go-utils/pathutil"
 )
 
+// ConfigModel ...
+type ConfigModel struct {
+	SetupVersion          string
+	LastPluginUpdateCheck int64
```
Comment (godrei) [inline configs/configs.go]: use time.Time instead of int

File: plugins/models.go [hunk 6]
```diff
@@ -17,6 +17,7 @@ type PluginRoute struct {
 	CommitHash   string `yaml:"commit_hash"`
 	Executable   string `yaml:"executable"`
 	TriggerEvent string `yaml:"trigger"`
+	Update       string `yaml:"update"`
```
Comment (godrei) [inline plugins/models.go]: LatestDiscoveredVersion / LatestKnownVersion / LatestAvailableVersion

File: plugins/models.go [hunk 7]
```diff
@@ -11,12 +11,13 @@ const (
...
-	CommitHash   string `yaml:"commit_hash"`
-	Executable   string `yaml:"executable"`
-	TriggerEvent string `yaml:"trigger"`
+	Name                   string `yaml:"name"`
+	Source                 string `yaml:"source"`
+	Version                string `yaml:"version"`
+	CommitHash             string `yaml:"commit_hash"`
+	Executable             string `yaml:"executable"`
+	TriggerEvent           string `yaml:"trigger"`
+	LatestAvailableVersion string `yaml:"last_available_version"`
```
Comment (viktorbenei) [inline plugins/models.go]: `latest_...`

**Pull request merged**